lib/engine/game/g_1848/game.rb

```diff
@@ -492,8 +533,18 @@ def init_stock_market
                                  multiple_buy_types: self.class::MULTIPLE_BUY_TYPES)
         end
 
+        def init_share_pool
+          G1848::SharePool.new(self)
+        end
+
+        def operating_order
+          boe, others = @corporations.select(&:floated?).sort.partition { |c| c.type == :bank }
+          boe + others
+        end
+
         def upgrades_to?(from, to, _special = false, selected_company: nil)
           return %w[5 6 57].include?(to.name) if (from.hex.tile.label.to_s == 'K') && (from.hex.tile.color == 'white')
+          return ['241'].include?(to.name) if selected_company && selected_company.sym == 'P3'
```

Comment:
if selected_company&.sym == 'P3'

Reply:
that's what happens when you start developing in a language you're not familiar with. 
I had no ides what &. meant.  Now i know Thanks!